lm_eval/tasks/codexglue_text_to_text.py

```diff
@@ -0,0 +1,117 @@
+
+class CodexglueTextToText(Task):
+
+    DATASET_PATH = "code_x_glue_tt_text_to_text"
+    DATASET_NAME = None
+
+    def __init__(self, translation_task):
+        self.DATASET_NAME = translation_task
+        stop_words = ["\n"]
+        requires_execution = False
+        super().__init__(stop_words, requires_execution)
+
+    def get_dataset(self):
+        """Returns dataset for the task or an iterable of any object, that get_prompt can handle"""
+        return self.dataset["test"]
+
+    def fewshot_examples(self):
+        """Loads and returns the few-shot examples for the task if they exist."""
+        with open(
+            "lm_eval/tasks/few_shot_examples/codexglue_text_to_text_few_shot_prompts.json",
+            "r",
+        ) as file:
+            examples = json.load(file)
+        return examples
+
+    @staticmethod
+    def two_shot_prompt(entry, text, examples, language):
+        """Two shot prompt format as source & target language documentation"""
+        prompt = f"\n{language.title()}:\n{examples['source1']}\
+                   \nEnglish:\n{examples['target1']}\
+                   \n{language.title()}:\n{examples['source2']}\
+                   \nEnglish:\n{examples['target2']}\
+                   \n{language.title()}:\n{text}\
+                   \nEnglish:\n"
+        return entry + prompt
+
+    def get_prompt(self, doc):
+        """Builds the prompt for the LM to generate from."""
+        language = SOURCE_LANG[self.DATASET_NAME]
+        text = doc["source"]
+        entry = f"Translate the following documentation from {language.title()} to English:\n"
+        examples = self.fewshot_examples()
+        examples = examples[language]
+        prompt = self.two_shot_prompt(entry, text, examples, language)
+        return prompt
+
+    def get_reference(self, doc):
+        """Builds the reference solution for the doc (sample from the test dataset)."""
+        return doc["target"].strip()
+
+    def postprocess_generation(self, generation, idx):
+        """Defines the postprocessing for a LM generation.
+        :param generation: str
+            code generation from LM
+        :param idx: int
+            index of doc in the dataset to which the generation belongs
+            (not used for this task)
+        """
+        output = generation.split("\nEnglish:\n", 3)[-1].strip()
+        return output
+
+    def process_results(self, generations, references):
+        """Takes the list of LM generations and evaluates them against ground truth references,
+        returning the metric for the generations.
+        :param generations: list(list(str))
+            list of lists containing generations
+        :param references: list(str)
+            list of str containing references
+        """
+        bleu = load("bleu")
+        gens = [gen[0] for gen in generations]
+        results = bleu.compute(
+            references=references, predictions=gens, max_order=4, smooth=True
+        )["bleu"]
```

Comment:
We changed the way the metric is returned to include its name and other details in this case for BLEU
```suggestion
        )
```